iothub/device/tests/Pipeline/RetryDelegatingHandlerImplicitOpenTests.cs

```diff
@@ -21,26 +20,26 @@ public class RetryDelegatingHandlerImplicitOpenTests
         [TestMethod]
         public async Task OpenAsyncNextCompletedSubjIsOpen()
         {
-            var contextMock = Substitute.For<PipelineContext>();
+            var contextMock = new PipelineContext();
             contextMock.ConnectionStatusChangeHandler = (connectionStatusInfo) => { };
-            var nextHandlerMock = Substitute.For<IDelegatingHandler>();
-            nextHandlerMock.OpenAsync(Arg.Any<CancellationToken>()).Returns(t => Task.CompletedTask);
+            var nextHandlerMock = new Mock<IDelegatingHandler>();
+            nextHandlerMock.Setup(x => x.OpenAsync(It.IsAny<CancellationToken>())).Returns(() => Task.CompletedTask);
 
             var cancellationToken = new CancellationToken();
-            var sut = new RetryDelegatingHandler(contextMock, nextHandlerMock);
+            var sut = new RetryDelegatingHandler(contextMock, nextHandlerMock.Object);
             await sut.OpenAsync(cancellationToken).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task ImplicitOpenSubjWasNotOpenSubjIsOpen()
```

Comment:
@brycewang-microsoft does this unit test even make sense in V2 with explicit open required?

Reply:
I think the test was updated and the test name wasn't fixed to match.

Reply:
> `NoImplicitOpenByClientMethods`?
"No implicit open" seems not what this test is doing here as it calls `OpenAsync()`, `WaitForTransportClosedAsync` and then `OpenAsync()` again. 

I notice this [comment ](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/device/src/Pipeline/RetryDelegatingHandler.cs#L386) so this test might be for an unsupported behavior?

Reply:
I think that comment is out of date. I think @timtay-microsoft did the work to enable open/close/open.

In the test here, we call open, then we call 2 methods, and in the end ensure that openasync was called only once (not 3 times). That seems to validate that there is no implicit open.